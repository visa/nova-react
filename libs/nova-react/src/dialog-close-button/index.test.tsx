/**
 *              Â© 2025 Visa
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **/
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import DialogCloseButton from '.';

describe('DialogCloseButton', () => {
  it('should render defaults correctly', async () => {
    const { container } = render(<DialogCloseButton />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container).toMatchSnapshot();
    expect(container.firstElementChild?.className).toBe(
      'v-button v-button-small v-button-tertiary v-button-icon v-button-subtle -v-mt-20 -v-ml-12 -v-mr-18'
    );
    expect(container.firstElementChild?.tagName).toBe('BUTTON');
  });

  it('should allow custom classNames', () => {
    const { container } = render(<DialogCloseButton className="test-class" />);
    expect(container.firstElementChild?.className).toBe(
      'v-button v-button-small v-button-tertiary v-button-icon v-button-subtle -v-mt-20 -v-ml-12 -v-mr-18 test-class'
    );
  });

  it('should render with aria-label', () => {
    const { container } = render(<DialogCloseButton ariaLabel="test-aria-label" />);
    expect(container.firstElementChild?.getAttribute('aria-label')).toBe('test-aria-label');
  });

  it('should allow custom elements', () => {
    const { container } = render(<DialogCloseButton element={<a />} />);
    expect(container.firstElementChild?.tagName).toBe('A');
  });

  it('should permeate basic props', () => {
    const { container } = render(<DialogCloseButton aria-label="test-aria-label" id="test-id" role="menu" />);
    expect(container.firstElementChild?.getAttribute('aria-label')).toBe('test-aria-label');
    expect(container.firstElementChild?.getAttribute('id')).toBe('test-id');
    expect(container.firstElementChild?.getAttribute('role')).toBe('menu');
  });

  it('should allow for child components', () => {
    const { container } = render(<DialogCloseButton>Test Child</DialogCloseButton>);
    expect(container.firstElementChild?.firstChild?.textContent).toBe('Test Child');
  });
});
