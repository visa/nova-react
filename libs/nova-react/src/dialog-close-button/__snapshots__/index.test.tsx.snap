// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogCloseButton should render defaults correctly 1`] = `
<div>
  <button
    aria-label="Close"
    class="v-button v-button-small v-button-tertiary v-button-icon v-button-subtle -v-mt-20 -v-ml-12 -v-mr-18"
  >
    <svg
      aria-hidden="true"
      class="v-icon v-icon-visa v-icon-tiny v-icon-close"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-primary"
        d="M8.00001 9.4149L13.9498 15.3646L15.364 13.9504L9.41423 8.00069L15.364 2.05094L13.9498 0.636723L8.00001 6.58647L2.05026 0.636719L0.636047 2.05093L6.5858 8.00069L0.636054 13.9504L2.05027 15.3646L8.00001 9.4149Z"
      />
      <defs>
        <clippath>
          <rect
            height="16"
            transform="translate(0 0.000488281)"
            width="16"
          />
        </clippath>
      </defs>
    </svg>
  </button>
</div>
`;
